### PR TITLE
feat: upgrade to gatsby v5

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Upgrades Gatsby and dependencies to V5

Builds on top of #1292

### CODE REVIEW

**Comment by Urigo** [comment]
@wardpeet I've just merged https://github.com/graphql/graphql.github.io/pull/1292
Can you rebase and we'll run the build again?

Also, with the last changes on this PR, the log displayed this:
```
7:06:46 PM: ---------------------------------------------------------------------DEPRECATION NOTICE: Builds using the Xenial build image will fail after November 15th, 2022.The build image for this site uses Ubuntu 16.04 Xenial Xerus, which is no longer supported.All Netlify builds using the Xenial build image will begin failing in the week of November 15th, 2022.To avoid service disruption, please select a newer build image at the following link:https://app.netlify.com/sites/graphql-org/settings/deploys#build-image-selectionFor more details, visit the build image migration guide:https://answers.netlify.com/t/please-read-end-of-support-for-xenial-build-image-everything-you-need-to-know/68239---------------------------------------------------------------------
7:07:04 PM: build-image version: e3a92d323f65dadd9483189ed53791424b7b61a7 (xenial)
7:07:04 PM: build-image tag: v3.15.1
7:07:04 PM: buildbot version: 0a9525e006898c513bdd79a097f61c04604a28eb
7:07:04 PM: Fetching cached dependencies
7:07:04 PM: Starting to download cache of 438.7MB
7:07:07 PM: Finished downloading cache in 2.469313426s
7:07:07 PM: Starting to extract cache
7:07:11 PM: Finished extracting cache in 4.07855348s
7:07:11 PM: Finished fetching cache in 6.611417414s
7:07:11 PM: Starting to prepare the repo for build
7:07:11 PM: Preparing Git Reference pull/1293/head
7:07:12 PM: Parsing package.json dependencies
7:07:13 PM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'public' versus '/' in the Netlify UI
7:07:13 PM: Different build command detected, going to use the one specified in the Netlify configuration file: 'yarn build' versus '' in the Netlify UI
7:07:13 PM: Section completed: initializing
7:07:13 PM: Starting build script
7:07:13 PM: Installing dependencies
7:07:13 PM: Python version set to 2.7
7:07:13 PM: Started restoring cached node version
7:07:14 PM: Finished restoring cached node version
7:07:14 PM: Attempting node version '18.12.1' from .node-version
7:07:15 PM: Downloading and installing node v18.12.1...
7:07:15 PM: Downloading https://nodejs.org/dist/v18.12.1/node-v18.12.1-linux-x64.tar.xz...
7:07:15 PM: Computing checksum with sha256sum
7:07:15 PM: Checksums matched!
7:07:17 PM: node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by node)node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by node)node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)nvm is not compatible with the npm config "prefix" option: currently set to ""
7:07:17 PM: Run `nvm use --delete-prefix v18.12.1` to unset it.
7:07:17 PM: Creating deploy upload records
7:07:17 PM: Failed to install node version '18.12.1'
7:07:17 PM: Build was terminated: Build script returned non-zero exit code: 1
7:07:17 PM: Failing build: Failed to build site
7:07:17 PM: Failed during stage 'building site': Build script returned non-zero exit code: 1 (https://ntl.fyi/exit-code-1)
7:07:18 PM: Finished processing build request in 13.38802278s
```

**Comment by wardpeet** [comment]
Maybe node 18 is not working on Netlify? 😬

**Comment by Urigo** [comment]
@wardpeet latest log
```
12:17:26 AM: ---------------------------------------------------------------------DEPRECATION NOTICE: Builds using the Xenial build image will fail after November 15th, 2022.The build image for this site uses Ubuntu 16.04 Xenial Xerus, which is no longer supported.All Netlify builds using the Xenial build image will begin failing in the week of November 15th, 2022.To avoid service disruption, please select a newer build image at the following link:https://app.netlify.com/sites/graphql-org/settings/deploys#build-image-selectionFor more details, visit the build image migration guide:https://answers.netlify.com/t/please-read-end-of-support-for-xenial-build-image-everything-you-need-to-know/68239---------------------------------------------------------------------
12:17:39 AM: build-image version: e3a92d323f65dadd9483189ed53791424b7b61a7 (xenial)
12:17:39 AM: build-image tag: v3.15.1
12:17:39 AM: buildbot version: 0a9525e006898c513bdd79a097f61c04604a28eb
12:17:39 AM: Fetching cached dependencies
12:17:39 AM: Starting to download cache of 662.0MB
12:17:46 AM: Finished downloading cache in 6.811634821s
12:17:46 AM: Starting to extract cache
12:17:52 AM: Finished extracting cache in 5.963017049s
12:17:52 AM: Finished fetching cache in 12.84753899s
12:17:52 AM: Starting to prepare the repo for build
12:17:52 AM: Preparing Git Reference pull/1293/head
12:17:53 AM: Parsing package.json dependencies
12:17:54 AM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'public' versus '/' in the Netlify UI
12:17:54 AM: Different build command detected, going to use the one specified in the Netlify configuration file: 'yarn build' versus '' in the Netlify UI
12:17:54 AM: Section completed: initializing
12:17:54 AM: Starting build script
12:17:54 AM: Installing dependencies
12:17:54 AM: Python version set to 2.7
12:17:55 AM: Started restoring cached node version
12:17:56 AM: Finished restoring cached node version
12:17:56 AM: Attempting node version '18.12.1' from .node-version
12:17:56 AM: Downloading and installing node v18.12.1...
12:17:57 AM: Downloading https://nodejs.org/dist/v18.12.1/node-v18.12.1-linux-x64.tar.xz...
12:17:57 AM: Computing checksum with sha256sum
12:17:57 AM: Checksums matched!
12:17:59 AM: node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by node)node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by node)node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)nvm is not compatible with the npm config "prefix" option: currently set to ""
12:17:59 AM: Run `nvm use --delete-prefix v18.12.1` to unset it.
12:17:59 AM: Failed to install node version '18.12.1'
12:17:59 AM: Creating deploy upload records
12:17:59 AM: Build was terminated: Build script returned non-zero exit code: 1
12:17:59 AM: Failing build: Failed to build site
12:17:59 AM: Failed during stage 'building site': Build script returned non-zero exit code: 1 (https://ntl.fyi/exit-code-1)
12:18:00 AM: Finished processing build request in 20.323191559s
```

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
       - uses: actions/checkout@v1
       - uses: actions/setup-node@v1
         with:
-          node-version: '16.13.0'
+          node-version: "18.12.1"
```
Comment (Urigo) [inline .github/workflows/CI.yml:12]: could it be related to the brackets?

**Comment by Urigo** [comment]
ok it was an old build image, I've updated it on Netlify and it worked.
the status didn't get updated back on GitHub but I've also reviewed the preview and it looks good
Merging

**Pull request merged**